bucket/bucket.go

```diff
@@ -4,31 +4,25 @@ import (
 	"net/http"
 	"strings"
 	"sync"
+	"github.com/mateusz/tempomat/lib/config"
 )
 
 type Bucket struct {
 	lowCreditThreshold float64
 	rate               float64
 	hashMaxLen         int
-	mutex              sync.RWMutex
```

Comment:
This is the canonical way of using sync.Mutex and sync.RWMutex in go, just embed it.